### PR TITLE
Verify snowflake

### PR DESCRIPTION
**Pull request by ahuang11**
## Summary

For verifying snowflake tasks
https://github.com/ahuang11/prefect-task-verification/blob/main/snowflake-tasks/snowflake.py

## Changes

Add missing docs for SnowflakeQueriesFromFile task, add guard against missing auth method, and add verification badges

## Importance

For verification purposes

## Checklist

This PR:

- [x] adds new tests (if appropriate)
- [ ] adds a change file in the `changes/` directory (if appropriate)
- [x] updates docstrings for any new functions or function arguments, including `docs/outline.toml` for API reference docs (if appropriate)

### CODE REVIEW

**Comment by ahuang11** [comment]
The linting seems unrelated
```
src/prefect/executors/dask.py:629: error: Function "dask.delayed.delayed" is not valid as a type
src/prefect/executors/dask.py:629: note: Perhaps you need "Callable[...]" or a callback protocol?
Found 1 error in 1 file (checked 289 source files)
```

File: docs/core/task_library/overview.md [hunk 1]
```diff
@@ -68,7 +68,7 @@ Tasks marked with the <Badge text="Verified" type="success" vertical="middle"></
   </tr>
   <tr>
     <td><img src="/logos/slack.png" height="128" width="128" style="max-height: 128px; max-width: 128px;"><a href="https://docs.prefect.io/api/latest/tasks/notifications.html#slacktask">Slack</a></td>
-    <td><img src="/logos/snowflake.png" height="128" width="128" style="max-height: 128px; max-width: 128px;"><a href="https://docs.prefect.io/api/latest/tasks/snowflake.html">Snowflake</a></td>
+    <td><td><Badge text="Verified" type="success"></Badge><img src="/logos/snowflake.png" height="128" width="128" style="max-height: 128px; max-width: 128px;"><a href="https://docs.prefect.io/api/latest/tasks/snowflake.html">Snowflake</a></td>
```
Comment (tpdorsey) [inline docs/core/task_library/overview.md:71]: I think you have too many td elements here?

File: src/prefect/tasks/snowflake/snowflake.py [hunk 2]
```diff
@@ -15,9 +17,9 @@ class SnowflakeQuery(Task):
              package documentation for details
         - user (str): user name used to authenticate
         - password (str, optional): password used to authenticate.
```
Comment (tpdorsey) [inline src/prefect/tasks/snowflake/snowflake.py:19]: As a general comment on docstrings, I'd love for us to move to correct sentence structure for argument descriptions: Capitalized first word unless we're referencing a literal. Period at end of sentence. In this case it's consistent as is and I'm okay with leaving it that way until further work is needed. Your call.

Comment (tpdorsey) [reply src/prefect/tasks/snowflake/snowflake.py:19]: To clarify, both styles are still reflected in task docstrings. I would like to move v1 docs to sentence style as it makes sense to do so. I am enforcing sentence style in v2.

File: src/prefect/tasks/snowflake/snowflake.py [hunk 3]
```diff
@@ -15,9 +17,9 @@ class SnowflakeQuery(Task):
...
-            password or private_key must be present
+            password or private_key must be present, unless authenticator is provided
```
Comment (tpdorsey) [inline src/prefect/tasks/snowflake/snowflake.py:20]: You can use markdown style ticks to format `private_key` as code throughout.

File: src/prefect/tasks/snowflake/snowflake.py [hunk 4]
```diff
@@ -129,11 +131,11 @@ def run(
                 takes snowflake AUTOCOMMIT parameter
             - cursor_type (SnowflakeCursor, optional): specify the type of database
                 cursor to use for the query, defaults to SnowflakeCursor
-        - authenticator (str, optional): type of authenticator to use for initiating
-            connection (oauth, externalbrowser...), refer to snowflake documentation
-            https://docs.snowflake.com/en/user-guide/python-connector-api.html#connect
-            for details, note that `externalbrowser` will only work in an environment
-            where a browser is available, default to None
+            - authenticator (str, optional): type of authenticator to use for initiating
+                connection (oauth, externalbrowser...), refer to snowflake documentation
```
Comment (tpdorsey) [inline src/prefect/tasks/snowflake/snowflake.py:135]: Capitalize Snowflake

File: src/prefect/tasks/snowflake/snowflake.py [hunk 5]
```diff
@@ -129,11 +131,11 @@ def run(
...
+                https://docs.snowflake.com/en/user-guide/python-connector-api.html#connect
```
Comment (tpdorsey) [inline src/prefect/tasks/snowflake/snowflake.py:136]: You can format the link as a markdown link. I would use something like this:

"refer to Snowflake [Python Connector API](https://docs.snowflake.com/en/user-guide/python-connector-api.html#connect) documentation for details. Note that..."

File: src/prefect/tasks/snowflake/snowflake.py [hunk 6]
```diff
@@ -311,11 +306,11 @@ def run(
...
```
Comment (tpdorsey) [inline src/prefect/tasks/snowflake/snowflake.py:311]: See comment above about formatting the link and tweaking the sentence a bit.

**Review by tpdorsey — APPROVED** [review]
Looks good. There are no critical doc issues here, but I have some comments about preferred style.

**Review by desertaxle — APPROVED** [review]

**Comment by madkinsz** [comment]
@ahuang11 can you rebase master to get the fixed type checks?

**Comment by ahuang11** [comment]
I think this is ready to go

**Comment by ahuang11** [comment]
@madkinsz let me know if there's anything else for this PR

**Pull request merged**